com.civicrm.hrjobroles/js/src/job-roles/directives/loader.js

```diff
@@ -0,0 +1,55 @@
+define([
+  'job-roles/directives/directives'
+], function(directives) {
+  'use strict';
+
+  directives.directive('jobrolesLoader', ['$rootScope', '$log', function($rootScope, $log) {
```

Comment:
This is not good. You have basically copy-pasted the same directive that we have on job contract. It's highly unmaintainable and contributes to the bloating of the code base.

There is an extension, [org.civicrm.reqangular](https://github.com/civicrm/civihr/tree/staging/org.civicrm.reqangular) whose job is to be a repo of all the common modules (directives, filters, etc) that might be shared between extension. We already have a [`loading`](https://github.com/civicrm/civihr/blob/staging/org.civicrm.reqangular/src/common/directives/loading.js) directive that displays a spinner. See if you can use that. If for whatever reason you cannot, then you need to see why and to integrate `loading` with this `jobrolesLoader`